### PR TITLE
routing navigation done

### PR DESCRIPTION
**Pull request by sauravces**
next project structure understanding, nested route , dynamic route, nested dynamic route, catch all segments, private folders , route groups , layouts, nested layouts , group layout, routing meta data , component navigation, Links are covered in this POC . but not focused on styling.

### CODE REVIEW

File: src/app/(auth)/(with-auth-layout)/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export default function AuthLayout({children}:{
+    children:React.ReactNode;
+}){
+return(
+    <>
+    <h2>Inner Layout</h2>
+    {children}
+
```
Comment (vaishali3012) [inline src/app/(auth)/(with-auth-layout)/layout.tsx]: line break is not needed

File: src/app/(auth)/layout.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  { name: "Forget Password", href: "/forget-password" },
+];
+
+export default function AuthLayout({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  const pathname = usePathname();
+  const [input,setInput]=useState("");
```
Comment (vaishali3012) [inline src/app/(auth)/layout.tsx]: a line break is needed below

File: src/app/(auth)/layout.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  return (
+    <div>
```
Comment (vaishali3012) [inline src/app/(auth)/layout.tsx]: use fragment if div is not needed

File: src/app/_lib/page.tsx [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export default function Lib() {
+    return (
+      <h1>This is Lib Folder</h1>
```
Comment (vaishali3012) [inline src/app/_lib/page.tsx]: Why this is needed ?

File: src/app/about/page.tsx [hunk 5]
```diff
@@ -0,0 +1,10 @@
+export const metadata={
+    title:"About",
+
```
Comment (vaishali3012) [inline src/app/about/page.tsx]: line break not needed

File: src/app/(auth)/(with-auth-layout)/layout.tsx (hunk 1 above)
Comment (aashikvilla) [reply src/app/(auth)/(with-auth-layout)/layout.tsx]: you can use prettier to avoid such formatting issues

**Pull request merged**